compiler/src/iree/compiler/Dialect/Flow/Transforms/VerifyInputLegality.cpp

```diff
@@ -17,12 +17,40 @@ namespace iree_compiler {
 namespace IREE {
 namespace Flow {
 
+static llvm::cl::opt<bool> clVerifyNoF64(
+    "iree-flow-verify-no-f64",
+    llvm::cl::desc("Verify that there are no ops that use f64."),
+    llvm::cl::init(true));
+
+// TODO: This helper function should be in MLIR.
+static Type getTensorOrVectorElementType(Type type) {
```

Comment:
https://github.com/llvm/llvm-project/blob/e09a1b51ab04b0d6d1d4ba5425f75a04d6b3a13d/mlir/include/mlir/IR/TypeUtilities.h#L31 ?

Reply:
unless I'm missing something getElementTypeOrSelf doesn't look recursively into the element type, e.g. for tensor<vector<f64>> it will stop early (on vector).
Basically I wish https://github.com/llvm/llvm-project/blob/e09a1b51ab04b0d6d1d4ba5425f75a04d6b3a13d/mlir/lib/IR/Operation.cpp#L876 was exposed in MLIR,
we can switch to getElementTypeOrSelf but then we might miss some cases ...

Reply:
at this level there shouldn't be any vector types so getElementTypeOrSelf should be fine

Reply:
ok, done